src/Presentation/Console/Command/QaAutomationCommand.php

```diff
@@ -0,0 +1,171 @@
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\ContextRepositoryInterface $contextRepository
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\ProjectSettingRepositoryInterface $projectSettingRepository
+     * @param \SprykerSdk\Sdk\Core\Application\Service\ContextFactory $contextFactory
+     * @param \SprykerSdk\Sdk\Presentation\Console\Command\OptionExtractor $optionExtractor
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\Repository\TaskRepositoryInterface $taskRepository
+     */
+    public function __construct(
+        TaskExecutor $taskExecutor,
+        ProjectWorkflow $projectWorkflow,
+        ContextRepositoryInterface $contextRepository,
+        ProjectSettingRepositoryInterface $projectSettingRepository,
+        ContextFactory $contextFactory,
+        OptionExtractor $optionExtractor,
+        TaskRepositoryInterface $taskRepository
+    ) {
+        $this->taskRepository = $taskRepository;
+        try {
+            $taskIds = $projectSettingRepository->getOneByPath(static::TASKS_SETTING_KEY)->getValues();
+            $this->task = $this->fillTask(
+                $this->taskRepository->findByIds($taskIds),
+            );
+            $taskOptions = $optionExtractor->extractOptions($this->task);
+        } catch (TableNotFoundException $e) {
+            $this->setHidden(true);
+            $taskOptions = [];
+        }
+
+        parent::__construct(
+            $taskExecutor,
+            $projectWorkflow,
+            $contextRepository,
+            $projectSettingRepository,
+            $contextFactory,
+            $taskOptions,
+            static::DESCRIPTION,
+            static::COMMAND_NAME,
+        );
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if ($this->projectWorkflow->getProjectWorkflows()) {
+            $output->writeln('<error>Your project has initialized workflow. Follow the workflow. See details for `sdk:workflow:run` command.</error>');
+
+            return static::FAILURE;
+        }
+
+        $context = $this->buildContext($input);
+        $context->setTask($this->task);
+        $context = $this->taskExecutor->execute($context);
+        $this->writeContext($input, $context);
+        $this->writeFilteredMessages($output, $context);
+
+        return $context->getExitCode();
+    }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Entity\TaskInterface> $tasks
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\TaskInterface
+     */
+    protected function fillTask(array $tasks): TaskInterface
+    {
+        $commands = [];
+        $placeholders = [];
+
+        foreach ($tasks as $task) {
+            foreach ($task->getCommands() as $command) {
+                $commands[] = new Command(
```

Comment:
Should be extracted to a separate method to split responsibilities

Reply:
fixed